salt/minion.py

```diff
@@ -422,6 +422,41 @@ class MinionBase(object):
     def __init__(self, opts):
         self.opts = opts
 
+    def gen_modules(self, initial_load=False):
+        '''
+        Tell the minion to reload the execution modules
+
+        CLI Example:
+
+        .. code-block:: bash
+
+            salt '*' sys.reload_modules
+        '''
+        self.opts['pillar'] = salt.pillar.get_pillar(
+            self.opts,
+            self.opts['grains'],
+            self.opts['id'],
+            self.opts['saltenv'],
+            pillarenv=self.opts.get('pillarenv'),
+        ).compile_pillar()
+
+        self.utils = salt.loader.utils(self.opts)
+        self.functions = salt.loader.minion_mods(self.opts, utils=self.utils)
+        self.serializers = salt.loader.serializers(self.opts)
+        self.returners = salt.loader.returners(self.opts, self.functions)
+        self.proxy = salt.loader.proxy(self.opts, self.functions, self.returners, None)
+        # TODO: remove
```

Comment:
Should you remove this now, or could you provide a better explanation here about exactly why we want to remove it?

Reply:
Aaaand  now I see that this block was just moved from somewhere else.

Which just validates my reasoning :trollface:

Reply:
I don't know the history of this line. What I did was move the method from the `Minion` class to the `MinionBase` class. So that it's available to all of the `MinionBase` subclasses. I didn't remove the commented out line so that whomever commented out can address it.

Having said that, I don't think I ended up using this method anywhere other than on the `Minion` so it's probably best to put it back where it was for now.

Reply:
Actually, I moved it from `SMinion` to `MinionBase` so that instances of `Minion` have access to it.